### PR TITLE
Enable using summary2 for analysis

### PR DESCRIPTION
**Pull request by keertip**
Refactor to use summary2 as sometimes FunctionType's are not backed by elements, we have use the type to get the information.

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -6956,7 +6873,7 @@ class PackageBuilder {
       //                   many AnalysisDrivers
       // TODO(jcollins-g): make use of DartProject isApi()
       // TODO(keertip): Use summary 2 after fixing #2017
-      AnalysisDriver.useSummary2 = false;
+      AnalysisDriver.useSummary2 = true;
```
Comment (scheglov) [inline lib/src/model.dart]: Remove this?
It is `true` by default, and in general you should not access private APIs :-)

Comment (keertip) [reply lib/src/model.dart]: done.

File: lib/src/element_type.dart [hunk 2]
```diff
@@ -5,23 +5,30 @@
 /// The models used to represent Dart code.
 library dartdoc.element_type;
 
+import 'dart:collection';
```
Comment (devoncarew) [inline lib/src/element_type.dart]: organize imports?

File: lib/src/element_type.dart [hunk 3]
```diff
@@ -83,31 +93,104 @@ abstract class ElementType extends Privacy {
...
+    }
     return name;
   }
 
   @override
-  String get nameWithGenerics => name;
+  String get name => type.name ?? '';
+}
+
+// A FunctionType that does not have an underpinning Element.
```
Comment (devoncarew) [inline lib/src/element_type.dart]: convert to a dartdoc comment?

File: lib/src/element_type.dart (hunk 2 above)
Comment (keertip) [reply lib/src/element_type.dart]: done.

File: lib/src/element_type.dart (hunk 3 above)
Comment (keertip) [reply lib/src/element_type.dart]: done.

**Comment by keertip** [comment]
Fixes #2017

**Pull request merged**